flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/FileStoreReadImpl.java

```diff
@@ -81,14 +87,42 @@ public FileStoreRead withValueProjection(int[][] projectedFields) {
     }
 
     @Override
-    public RecordReader createReader(BinaryRowData partition, int bucket, List<SstFileMeta> files)
+    public RecordReader createReader(BinaryRowData partition, int bucket, List<DataFileMeta> files)
             throws IOException {
+        switch (writeMode) {
+            case APPEND_ONLY:
+                return createAppendOnlyReader(partition, bucket, files);
+
+            case CHANGE_LOG:
+                return createLSMReader(partition, bucket, files);
+
+            default:
+                throw new UnsupportedOperationException("Unknown write mode: " + writeMode);
+        }
+    }
+
+    private RecordReader createAppendOnlyReader(
+            BinaryRowData partition, int bucket, List<DataFileMeta> files) throws IOException {
+        Preconditions.checkArgument(
+                !dropDelete, "Cannot drop delete message for append-only table.");
+
+        SstFileReader fileReader = sstFileReaderFactory.create(partition, bucket);
```

Comment:
*TODO*

Currently, both AppendOnlyReader and LSMReader are using the `SstFileReader`, while the `sst` concept should not be appeared in the append-only read path. So it's better to move this into a totally new `DataFileReader ` ?

Reply:
+1

Reply:
I think we can start renaming in a separate JIRA/PR.

Reply:
Changing very much classes can cause many conflicts, for example with https://github.com/apache/flink-table-store/pull/107
Let's get this refactored as early as possible, I'm advancing it here: https://github.com/apache/flink-table-store/pull/108